core/src/main/java/org/apache/gravitino/relation/Relation.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.relation;
+
+public interface Relation {
+
+  enum Type {
```

Comment:
It's okay for me to introduce the concept of `relation`, I have thought about it and have the following point: 
1. Is this relationship directed or not? I recall that we need bidirectional relational mapping for tag and entity.
2. Currently, schema-table, and catalog-schema can also be classified as a relation, perhaps we could call it the parent-child relation or something similar.

Reply:
1. It's directed and we should judge the source type and dest type.
2. I hesitate to add the parent-child relation.